File: src/procedures/MultiProcedure.java
package procedures;

import led.LEDDataBundle;
import led.LEDStripManager;
import led.ProcedureBundleFields;

public class MultiProcedure extends Procedure {

	Procedure[] mProcedures;
	
	public MultiProcedure(LEDDataBundle _bundle) {
		super(_bundle);
		
		//TODO Re-enable
		//mProcedures = (Procedure[]) _bundle.get(ProcedureBundleFields.SUB_BUNDLE);
	}
	
	@Override
	public void start() {
		for (Procedure procedure : mProcedures) {
			procedure.start();
			//mSteps = procedure.mSteps > mSteps ? procedure.mSteps : mSteps; 
		}
		mSteps = 60;
	}

	@Override
	void update() {
		mStep++;
		for (Procedure procedure : mProcedures) {
			procedure.update();
		}
		
		if (mStep >= mSteps) {
			finishProcedure();
		}
	}
}

File: src/main/java/procedures/MultiProcedure.kt
package procedures

import com.github.iamniklas.liocore.led.LEDDataBundle
import com.github.iamniklas.liocore.procedures.Procedure

class MultiProcedure(_bundle: LEDDataBundle) : Procedure(_bundle) {
    lateinit var mProcedures: Array<Procedure>
    override fun start() {
        for (procedure in mProcedures) {
            procedure.start()
            //mSteps = procedure.mSteps > mSteps ? procedure.mSteps : mSteps; 
        }
        steps = 60
    }

    public override fun update() {
        step++
        for (procedure in mProcedures) {
            procedure.update()
        }
        if (step >= steps) {
            finishProcedure()
        }
    }
}